test/pdevs_dynamic_coordinator_test.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2018, Laouen M. L. Belloli
+ * Carleton University, Universidad de Buenos Aires
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+#include <cadmium/modeling/dynamic_coupled.hpp>
+#include <cadmium/engine/pdevs_dynamic_coordinator.hpp>
+#include <cadmium/logger/common_loggers.hpp>
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/basic_model/generator.hpp>
+#include <cadmium/modeling/coupled_model.hpp>
+#include <cadmium/modeling/dynamic_atomic.hpp>
+#include <cadmium/modeling/dynamic_model_translator.hpp>
+
+BOOST_AUTO_TEST_SUITE( pdevs_coordinator_test_suite )
```

Comment:
Please rename to not having issues finding out which one of the 2 is failing (dynamic or not)

Reply:
They are duplicated, but I think we shouldn't rename the function, if one fails, the bug is replicated in the other and should be corrected. 

We should have a file with common functions like this one to allow a better separation of cadmium and cadmium dynamic.